src/python/WMCore/MicroService/MSPileup/MSPileupData.py

```diff
@@ -79,6 +79,40 @@ def customDID(pname):
     return pname
 
 
+def updateTransitionRecord(doc, userDN, logger):
+    """
+    Update transition record based on the following logic:
+    - check previous fraction value and add new transition record if it is
+    - differ from current one or does not exist
+      - first use-case: differ from container fraction of record
+      - second use-case: differ from last transition container fraction record
+    - https://gist.github.com/amaltaro/b4f9bafc0b58c10092a0735c635538b5
+
+    :param doc: MSPileup record
+    :param userDN: user DN (string)
+    :param logger: logger to use
+    :return: nothing (document transition is updated in place)
+    """
+    fraction = doc.get('containerFraction', 1.0)
+    transition = doc.get('transition', [])
+    prevTranRecord = transition[-1]
+    if not userDN:
+        userDN = prevTranRecord.get('DN', '')
+        if not userDN:
+            msg = f'Last transition record {prevTranRecord} does not have DN attribute'
```

Comment:
In standard conditions and free of bug, we should never reach this code. So I would suggest to simply remove this unneeded complexity.

Reply:
ok, fair